uvicorn/_async_agnostic/http11/keepalive.py

```diff
@@ -0,0 +1,52 @@
+from contextlib import AsyncExitStack
+from typing import Optional
+
+from uvicorn.config import Config
+
+from ..backends.auto import AutoBackend
+from ..backends.base import TaskHandle, TaskStatus
+from .connection import HTTP11Connection
+
+
+class KeepAlive:
```

Comment:
I find it nice that this implementation uses a separate `KeepAlive` component that's somewhat testable in isolation. In `master` the keep-alive logic is entagled with the main server code, and there's low-level logic around managing asyncio tasks. This kind of isolated component makes it a bit easier to understand what's going on, I believe.